src/mon/Elector.cc

```diff
@@ -754,6 +755,16 @@ void Elector::notify_rank_removed(int rank_removed)
       }
     }
   }
+  // Case where we are removing the highest rank monitor, we only care 
+  // about when the rank is in dead_pinging, which means
+  // it was shutted down, so we can erase it from dead_pinging.
+  // We don't care in the case where rank_removed 
+  // is still in live_pinging since it can be added back since it is not dead.
+  if (rank_removed == paxos_size()){
```

Comment:
This won't work since rank start from 0 to N
paxos_size = N+1
change to something like:
monmap->get_rank(monmap->ranks.back());